lib/tpm2_capability.c

```diff
@@ -33,38 +33,97 @@
 
 #include <tss2/tss2_sys.h>
 
+#include <string.h>
+
 #include "log.h"
 #include "tpm2_util.h"
 
 #include "tpm2_capability.h"
 
+static int tpm2_get_property_data_size(TPM2_CAP capability) {
+    int size;
+
+    switch (capability) {
```

Comment:
switch and return sizeof the field in the subordinate union member.

```C
TPMU_CAPABILITIES dummy;
switch (capability) {
case TPM2_CAP_ALGS:
    return sizeof(dummy.algorithms.algProperties[0]);
case ...;
    return ...'
...
```
Now you never have to worry about the spec changing a type, it'll always work or not compile.